pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -451,8 +451,7 @@ func (ku *Upgrader) upgradeAgentScaleSets(ctx context.Context) error {
 			Translator: ku.Translator,
 		}
 
-		if err = transformer.NormalizeForVMSSScaling(ku.logger, templateMap); err != nil {
-			ku.logger.Errorf("unable to update template, error: %v.", err)
+		if err = transformer.NormalizeMasterResourcesForScaling(ku.logger, templateMap); err != nil {
```

Comment:
This function is to normalize the master resource, why are we calling it in the upgradeAgent function? If it applies to both, we should rename it (also it says for scaling and this is an upgrade, I know it was already like that before, but maybe it's a good opportunity to make the name reflect what it's actually used for...)

Reply:
The template actually contains master node at this point. So we have to normalize it to make it work, I'm not sure if there is an option to generate agent only template. If there are, that seems the right approach. 

The function only normalize master node, nothing for agent. VMSS doesn't need anything to normalize. It's not for "Scaling" though, I can rename it NormalizeMasterResourcesForUpdate(), but that's a breaking change technically since it's exported function. If you are OK, I can make the change.

Reply:
Got it. The template is the full template as far as I understand that's why it contains all the resources including master resources. What happens for a template that doesn't contain any masters (ie AKS)? Would this also work?

Reply:
yep, it will skip, the normalization only filter master node.

```
		// make sure this is only modifying the master vms
		if !strings.Contains(resourceName, "variables('masterVMNamePrefix')") {
			continue
		}
```